lib/msal-browser/src/controllers/StandardController.ts

```diff
@@ -1196,18 +1197,50 @@ export class StandardController implements IController {
     // #region Account APIs
 
     /**
-     * Returns all accounts that MSAL currently has data for.
-     * (the account object is created at the time of successful login)
-     * or empty array when no accounts are found
-     * @returns Array of account objects in cache
+     * Returns all the accounts in the cache that match the optional filter. If no filter is provided, all accounts are returned.
+     * @param accountFilter - (Optional) filter to narrow down the accounts returned
+     * @returns Array of AccountInfo objects in cache
      */
-    getAllAccounts(): AccountInfo[] {
+    getAllAccounts(accountFilter?: AccountFilter): AccountInfo[] {
         this.logger.verbose("getAllAccounts called");
         return this.isBrowserEnvironment
-            ? this.browserStorage.getAllAccounts()
+            ? this.browserStorage.getAllAccounts(accountFilter)
             : [];
     }
 
+    /**
+     * Returns the first account found in the cache that matches the account filter passed in.
+     * @param accountFilter
+     * @returns The first account found in the cache matching the provided filter or null if no account could be found.
+     */
+    getAccount(accountFilter: AccountFilter): AccountInfo | null {
+        this.logger.trace("getAccount called");
+        if (!accountFilter || Object.keys(accountFilter).length === 0) {
```

Comment:
nit: accountFilter is required so we don't need to check for truthiness
```suggestion
        if (Object.keys(accountFilter).length === 0) {
```

Reply:
I wonder if we should also validate the values? Something like `{username: ""}` would bypass this check

Reply:
You mean a non-empty, non-null value always?

Reply:
@tnorling  @sameerag I think empty username can technically be a valid search filter though. I think filter debugging can be left to the devs, but I'm still open to adding validation if my first assumption isn't true.

Reply:
Empty filters evaluate as true. So a filter of `{username: ""}` would match ALL accounts in the cache.